src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -242,23 +253,21 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
             return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider);
         }
 
-        private static void ThrowEntryPointNotfoundException(
+        private static void NotifyEntryPointNotfound(
             DefaultHost host,
             string applicationName,
             Exception innerException)
         {
             if (host.Project.Commands.Any())
             {
-                // Throw a nicer exception message if the command
-                // can't be found
-                throw new InvalidOperationException(
-                    string.Format("Unable to load application or execute command '{0}'. Available commands: {1}.",
+                Console.Error.WriteLine(string.Format("Unable to load application or execute command '{0}'. Available commands: {1}.",
```

Comment:
I'd prefer a (non-abstract) base class, it's less brittle. Something like `Microsoft.Dnx.Runtime.RuntimeException`. Either way we have to go through each exception and figure out what to do, may as well do it "right". When we have other exceptions to throw (like InvalidOperationException) we can wrap them in the InnerException of RuntimeException